packages/ui5-tooling-modules/lib/utils/WebComponentRegistry.js

```diff
@@ -390,6 +403,45 @@ class RegistryEntry {
 		}
 	}
 
+	/**
+	 * Some web components need additional patches to comply with UI5 framework requirements,
+	 * e.g. applying the LabelEnablement mixin
+	 *
+	 * Most things patched in this function should eventually be available generically in the custom elements metadta.
+	 *
+	 * @param {object} ui5metadata the UI5 metadata object
+	 */
+	#patchUI5Specifics(ui5metadata) {
+		const { tag } = ui5metadata;
+
+		// The label has a couple of specifics that are not fully reflected in the custom elements.
+		// This code harmonizes the Label with the original retrofit control wrapper.
+		// TODO: How to generalize this?
+		//       Is harmonizing with the retrofit library really correct?
+		if (tag === "ui5-label") {
+			// the ui5-label has as default slot, but no aggregations on the retrofit layer...
+			ui5metadata.aggregations = [];
+			// the "for" attribute is called "labelFor" in the retrofit...
+			ui5metadata.associations["labelFor"] = {
+				type: "sap.ui.core.Control",
+				multiple: false,
+				mapping: {
+					type: "property",
+					to: "for",
+				},
+			};
+			delete ui5metadata.properties["for"];
+
+			// Any "Label" control needs a special UI5-only interface
+			// Additionally, all controls implementing this interface must apply the "sap/ui/core/LabelEnablement" mixin on their class
+			// refer to "../templates/WrapperControl.hbs"
+			ui5metadata.interfaces.push("sap.ui.core.Label");
+		} else if (tag === "ui5-multi-input") {
+			// TODO: Multi Input needs to implement the functions defined in "sap.ui.core.ISemanticFormContent"...
+			ui5metadata.interfaces.push("sap.ui.core.ISemanticFormContent");
```

Comment:
I don't know yet what this actual does... need to ask Sebastian about it.

Reply:
I'd keep this for now, as the interface itself is not wrong. But we need to clarify what the implications are:
`Form` vs `SemanticForm`?

Reply:
Ok, anyhow we go ahead an merge this change since it brings some new valuable features to align with the older retrofit by adding the enabled and textDirection mapping and adding the LabelSupport.